### PR TITLE
Share location with other apps (PSG-242)

### PR DESCRIPTION
**Pull request by onurays**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Users can now open others' (or their own) live locations with other map applications by tapping the marker of the users. Shared location is the static location at the time users click to share button.

## Screenshots / GIFs

|Share location|Open with|
|-|-|
|share_location|share_location_2|

## Tests

- Share a live location
- Click on timeline item to open maximized map
- Click on user's marker on the map
- You should see the "share" action

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by onurays** [comment]
Thanks @Claire1817 for reporting bugs related to marker rendering when the user doesn't have an avatar. [#6594](https://github.com/vector-im/element-android/issues/6594) will be fixed in another PR.

File: vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        views = ViewLiveLocationMarkerPopupBinding.bind(contentView)
+
+        width = ViewGroup.LayoutParams.WRAP_CONTENT
+        height = ViewGroup.LayoutParams.WRAP_CONTENT
+        inputMethodMode = INPUT_METHOD_NOT_NEEDED
+        isFocusable = true
+        isTouchable = true
+        elevation = 8f
+        setBackgroundDrawable(ContextCompat.getDrawable(context, R.drawable.bg_live_location_marker_popup))
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt]: It works, thanks, done.

File: vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        contentView = inflater.inflate(R.layout.view_live_location_marker_popup, null, false)
+
+        views = ViewLiveLocationMarkerPopupBinding.bind(contentView)
+
+        width = ViewGroup.LayoutParams.WRAP_CONTENT
+        height = ViewGroup.LayoutParams.WRAP_CONTENT
+        inputMethodMode = INPUT_METHOD_NOT_NEEDED
+        isFocusable = true
+        isTouchable = true
+        elevation = 8f
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt]: Exactly, done.

File: vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    interface Callback {
+        fun onShareLocationClicked()
+    }
+
+    private val views: ViewLiveLocationMarkerPopupBinding
+
+    var callback: Callback? = null
+
+    init {
+        contentView = inflater.inflate(R.layout.view_live_location_marker_popup, null, false)
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt]: Fixed.

File: vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        setBackgroundDrawable(ContextCompat.getDrawable(context, R.drawable.bg_live_location_marker_popup))
+
+        contentView.setOnClickListener {
+            callback?.onShareLocationClicked()
+        }
+    }
+
+    fun show(anchorView: View) {
+        contentView.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED)
+        showAsDropDown(anchorView, -contentView.measuredWidth / 2, 0)
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapMarkerOptionsDialog.kt:57]: Added.

**Review by Claire1817 — APPROVED** [review]
thanks for the update :)

**Pull request merged**